open_instruct/grpo_fast.py

```diff
@@ -753,7 +753,8 @@ def train(
         to_device_inplace(collated_position_ids, self.device)
         to_device_inplace(collated_advantages, self.device)
         to_device_inplace(collated_response_masks, self.device)
-        accumulation_steps = math.ceil(len(collated_query_responses) / num_mini_batches - 0.5)
+        # accumulation steps should always be at least 1
+        accumulation_steps = max(math.ceil(len(collated_query_responses) / num_mini_batches - 0.5), 1)
```

Comment:
Is this the case where one device's `len(collated_query_responses)` is 0? Also what is this `0.5` in `num_mini_batches - 0.5`?